lib/checkfunctions.cpp

```diff
@@ -682,3 +682,108 @@ void CheckFunctions::copyElisionError(const Token *tok)
                 "Using std::move for returning object by-value from function will affect copy elision optimization."
                 " More: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-return-move-local");
 }
+
+void CheckFunctions::useStandardLibrary()
+{
+    if (!mSettings->severity.isEnabled(Severity::style))
+        return;
+
+    for (const Scope& scope: mTokenizer->getSymbolDatabase()->scopeList) {
+        if (scope.type != Scope::ScopeType::eFor)
+            continue;
+
+        const auto *tok = scope.bodyStart; // { token
+        const auto *forToken = tok->previous()->link()->previous(); // should be for
+
+        // for ( firstExpression ; secondExpression ; thirdExpression )
+        const Token *firstExpression  = forToken->next()->astOperand2()->astOperand1();
+        if (!firstExpression)
+            continue;
+        const auto *astOperand22 = forToken->next()->astOperand2()->astOperand2();
+        if (!astOperand22)   // #7050
+            continue;
+        const Token *secondExpression = astOperand22->astOperand1();
+        if (!secondExpression)
+            continue;
+        const Token *thirdExpression = astOperand22->astOperand2();
+        if (!thirdExpression)
+            continue;
+
+        // 1. we expect that idx variable will be initialized with 0
+        const auto* idxToken = firstExpression->astOperand1();
+        const auto* initVal = firstExpression->astOperand2();
+        if (!idxToken || !initVal || !initVal->isNumber() || initVal->str() != "0")
+            continue;
+        const auto idxVarId = idxToken->varId();
+
+        // 2. we expect that idx will be less of something
+        if (!secondExpression->isComparisonOp())
+            continue;
+        const auto& secondOp = secondExpression->str();
+
+        if (!(("<" == secondOp && secondExpression->astOperand1()->varId() == idxVarId) ||
```

Comment:
I think it would be a good idea to check that the other side is a fixed value.

Reply:
It could be also (const) variable or expression like `my_size*sizeof(T)`

Reply:
yeah.. ensure in any way you want that the value will be unchanged during the loop. Simplest is to check if it has a known value but that approach will have false negatives. If you don't ensure that the value is unchanged there can be false positives.

Reply:
The trick is that we are checking entire body of loop. From this perspective I can't imagine a (real) code which can be on the right side. Maybe my imagination is a really poor.